Sources/ImperialCore/FederatedServiceRouter.swift

```diff
@@ -30,20 +26,20 @@ package protocol FederatedServiceRouter: Sendable {
     /// The URL on the app that will redirect to the `authURL` to get the access token from the OAuth provider.
     var accessTokenURL: String { get }
 
-    /// The URL of the page that the user will be redirected to to get the access token.
-    func authURL(_ request: Request) throws -> String
-
+    /// The URL components of the page that the user will be redirected to to get the access token.
+    func authURLComponents(_ request: Request) throws -> URLComponents
```

Comment:
Changed authURL to authURLComponents so the FederatedServiceRouter protocol extensions could centralize more code. Like handling the error for converting components to a URL in one place instead of multiple providers. And a `state` query item can be  easily generated in the protocol extension, saved to the session and added to the components.